### PR TITLE
Update Locales json files with camelCase keys and interpolated strings

### LINKED ISSUE #2059: Replace `kebap-case` within brackets in locale files with `camelCase`
## Problem

We use `kebap-case` in the locale string interpolations that are replaced by `VueI18n`. These should be replaced with `camelCase` to prevent errors in `Nuxt3`'s version of i18n. 

## Description

The keys that use `kebap-case` should be replaced with keys with `camelCase`. Care should be taken to check if they are correctly interpolated in the components.
Before
`Shown because {feat-name} is {feat-state}`
After
`Shown because {featName} is {feat-State}`

## Alternatives

There might be a way to fix the handling of dashes by @nuxjs/i18n 8, but it's probably easier to just replace the strings (there are about 13 of them).

## Additional context

If you update to the version of nuxt i18n that's compatible with Nuxt 3 (8), you'll get the following errors:
```
 ERROR  [unplugin-vue-i18n]  Unterminated closing brace (error code: 7) in /src/locales/en.json                                                                   2:40:38 PM
  target message: Shown because {feat-name} is {feat-state}
  target message path: pref-page.explanation

  1  |  Shown because {feat-name} is {feat-state}
```

### PR DESCRIPTION
**Pull request by sajeremy**
## Fixes
Converted keys and interpolated string from kebap case to camelCase.

Fixes #2059  by @obulat 

## Description
_Edited to add:_ This PR has been split into several PRs by @obulat to make it easier to review.

Replaced json and json5 locale files so that the keys and interpolated strings are in camelCase.  Used the following script below to automate replacement for keys and interpolated strings while avoiding converting string values in objects.

```import fs from 'fs'
import JSON5 from 'json5'
import kebabToCamel from './kebabCamelFunc.js'

//Locales English
const enData = fs.readFileSync('./en.json5','utf8')
const enObj = JSON5.parse(enData)
const newEnObj = kebabToCamel(enObj)
const enJson5Str = JSON5.stringify(newEnObj, null, 2);
fs.writeFileSync('./output_en.json5', enJson5Str);

//Locales Arabic
const arData = fs.readFileSync('./ar.json','utf8')
const arObj = JSON.parse(arData)
const newArObj = kebabToCamel(arObj)
const arJsonStr = JSON.stringify(newArObj,null,2)
fs.writeFileSync('./output_ar.json', arJsonStr)

//Locales Spanish
const esData = fs.readFileSync('./es.json','utf8')
const esObj = JSON.parse(esData)
const newEsObj = kebabToCamel(esObj)
const esJsonStr = JSON.stringify(newEsObj,null,2)
fs.writeFileSync('./output_es.json', esJsonStr)

//Locales Russian
const ruData = fs.readFileSync('./ru.json','utf8')
const ruObj = JSON.parse(ruData)
const newRuObj = kebabToCamel(ruObj)
const ruJsonStr = JSON.stringify(newRuObj,null,2)
fs.writeFileSync('./output_ru.json', ruJsonStr)
```
```
export default function kebabToCamel(input) {
  const camelCaseObj = {}; 
  for (const [key, value] of Object.entries(input)) {
    const camelCaseKey = key.replace(/-([\w])/g, (match, letter) => letter.toUpperCase());
    if (typeof value === "string") {
      const camelCaseValue = value.replace(/\{([^\}]+)\}/g, (match, content) =>
        `{${content.replace(/-([\w])/g, (match, letter) => letter.toUpperCase())}}`
      );
      camelCaseObj[camelCaseKey] = camelCaseValue;
    } else if (typeof value === "object") {
      camelCaseObj[camelCaseKey] = kebabToCamel(value);
    } else {
      camelCaseObj[camelCaseKey] = value;
    }
  }
  return camelCaseObj;
}
```

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sajeremy** [comment]
Hi @obulat, I just updated the keys and interpolated strings in the 4 locales json files i found. I'm hoping to take a closer look at the vue components tomorrow to replace the kebap case in these files. Also happy to make any adjustments for this pull request if there are any error or conflicts you spot.

**Comment by sarayourfriend** [comment]
@obulat Do you know if changes to the keys like this is going to cause every single key changed to appear as a new string in Glotpress? I don't know it well enough to know if it is smart enough to notice that only the keys have changed while the strings are the same. If you don't know or aren't sure, can we verify with the translations folks? Otherwise, I fear this will cause a significant dip in our translation percentages and lose the hard work the translators have done so far.

**Comment by sajeremy** [comment]
Hi @sarayourfriend , I'm not sure if it answers your question or is related but I wanted to mention that I need to update the original description of this pull request as it only describes the changes I made to the locales files.  I also wrote a script to change files in the src folder that utilize keys in the locales files and changed the referenced text from kebap case to camel case as well so that we can reference the different language locales files. 

If there are other directories that I need to update I'm more than happy to make changes or abandon all together depending on what you and @obulat believe is the best approach.

Currently when running the CI/CD testing, I get a failure during the linting test (I thought I turned off my linting and prettier extensions) but trying to see if there are other causes that are causing the test failure.

**Comment by sarayourfriend** [comment]
Yup, gotcha. The question I'm asking Olga is about how Glotpress will interpret these changes. For context, the `en.json` file gets transformed into a POT file which is uploaded to WordPress.org's Glotpress instance. WordPress volunteers then translate the file there, which we download and then transform back into a format that our Vue app can understand.

The changes in this PR look fine to me, I don't think there's an issue with them on their own. My question is just meant to ask whether this particular change will cause the application the translators use to translate strings (Glotpress) to treat all the new _keys_ as distinct and new strings that need to be translated, accidentally throwing away the work the translators have done so far. I suspect Glotpress is smart enough to notice that the values haven't changed and reuse them, but I don't know for certain. It would be good to know what the effect of this will be on translators and the progress of their work before we merge it and to mitigate any adverse effects if we can (probably through working with the WP.org Glotpress folks to bulk-change keys, especially if we _need_ this change for the Nuxt 3 upgrade).

**Comment by obulat** [comment]
> Yup, gotcha. The question I'm asking Olga is about how Glotpress will interpret these changes. For context, the `en.json` file gets transformed into a POT file which is uploaded to WordPress.org's Glotpress instance. WordPress volunteers then translate the file there, which we download and then transform back into a format that our Vue app can understand.
> 
> The changes in this PR look fine to me, I don't think there's an issue with them on their own. My question is just meant to ask whether this particular change will cause the application the translators use to translate strings (Glotpress) to treat all the new _keys_ as distinct and new strings that need to be translated, accidentally throwing away the work the translators have done so far. I suspect Glotpress is smart enough to notice that the values haven't changed and reuse them, but I don't know for certain. It would be good to know what the effect of this will be on translators and the progress of their work before we merge it and to mitigate any adverse effects if we can (probably through working with the WP.org Glotpress folks to bulk-change keys, especially if we _need_ this change for the Nuxt 3 upgrade).

I checked GlotPress, and I don't think this will cause problems. The key used in `.po` files in GlotPress are the strings themselves, and our `keys` are set as `msgctxt` (context). Since the strings stay the same, they will still be translated the same way. See these two objects:
```
#: src/components/VSearchResultsGrid/VAllResultsGrid.vue:111
msgctxt "browse-page.search-form.audio"
msgid "audio"
msgstr "аудио"

msgctxt "error.audio"
msgid "audio"
msgstr "аудио"
```

**Review by obulat — CHANGES_REQUESTED** [review]
Thank you for your changes, this is a really great start, @sajeremy! I checked the PR out locally, and found that in addition to the changes inside `$t()` function, it also needs more complicated changes in other parts of the code. One of such cases is `use-pages` composable.

I've created a patch with all the changes I added locally here: https://gist.github.com/obulat/c27b9a7232f15ab76a840e8cb3f73fba
If you want, I can add those changes to this PR, @sajeremy

**Comment by sajeremy** [comment]
> Thank you for your changes, this is a really great start, @sajeremy! I checked the PR out locally, and found that in addition to the changes inside `$t()` function, it also needs more complicated changes in other parts of the code. One of such cases is `use-pages` composable.
> 
> I've created a patch with all the changes I added locally here: https://gist.github.com/obulat/c27b9a7232f15ab76a840e8cb3f73fba If you want, I can add those changes to this PR, @sajeremy

@obulat wow, that is awesome! I would definitely appreciate adding those changes to the PR and  will taking the time to learn from your solution. Aside from the changes you mentioned above, are there any other actions you would like me to complete for this PR?

**Comment by obulat** [comment]
> I would definitely appreciate adding those changes to the PR and will taking the time to learn from your solution. Aside from the changes you mentioned above, are there any other actions you would like me to complete for this PR?

Great! I've added the changes and also merged main into this branch. I don't think there's too much to learn from my solution, though :) It's more knowledge of the project and the code than a clever solution, I've opened all the pages to make sure there are no warnings about missing i18n keys left.

I am going to approve this PR as soon as the CI passes, and after there is one more approval, this PR can be merged 🚀 

Thank you so much for working on this, and welcome to Openverse!

File: frontend/src/components/VAudioTrack/VGlobalAudioTrack.vue [hunk 1]
```diff
@@ -184,10 +184,8 @@ export default defineComponent({
     const pause = () => activeAudio.obj.value?.pause()
 
     /* Timekeeping */
-    const message = computed<string | undefined>(() =>
-      activeMediaStore.message
-        ? i18n.t(`audio-track.messages.${activeMediaStore.message}`).toString()
-        : undefined
+    const message = computed<string | undefined>(
+      () => activeMediaStore.message ?? undefined
```
Comment (obulat) [inline frontend/src/components/VAudioTrack/VGlobalAudioTrack.vue:188]: `activeMediaStore` saves message already translated, so we don't need to use i18n here

**Review by obulat — APPROVED** [review]
@sajeremy, I had to split the PR into several PRs to make them easier to review. After one more approval, this PR will be merged.
Thank you for your contribution 🎉

**Review by zackkrida — APPROVED** [review]
All pages look good! I suspect this was _so much_ manual work, especially those test cases and the example locales.

**Pull request merged**